### PR TITLE
feat(ai): migrate AI observability from PostHog to Phoenix Cloud

### PR DESCRIPTION
**Pull request by JeffOtano**
## Summary

- Cuts AI tracing from PostHog over to Phoenix Cloud so we capture full raw turns (system prompt, training snapshot, tool args/results, assistant output) for debugging and eval.
- Makes `aiRun.runId` equal the actual Phoenix trace id — retries + fallback share one trace and `aiToolCalls` now joins to it via `runId`, `toolCallId`, `argsJson`, `resultPreview`.
- Adds the first usable eval workflow: shared scenarios, a PR-smoke npm script with pass thresholds, a Phoenix dataset sync, and a scaffolded nightly agent experiment (rubric + banned-phrase + LLM judge). PostHog stays wired for product analytics only.

## Changes

**Tracing cutover (`convex/ai/otel.ts`, `resilience.ts`, `chatProcessing.ts`)**
- Replaces the PostHog exporter + attribute translator with `register()` from `@arizeai/phoenix-otel`.
- `runInRunSpan` wraps every chat/approval-continuation turn so retries inherit the trace context; `span.spanContext().traceId` becomes `aiRun.runId`.
- Flips `recordInputs`/`recordOutputs` to `true` and enriches telemetry metadata (`runId`, `threadId`, `userId`, `provider`, `source`, `environment`, `release`, `promptVersion`, `hasImages`, `isByok`). BYOK + Tonal sanitization upstream is unchanged.

**`RunAccumulator` gap-fill (`convex/ai/runTelemetry.ts`)**
- Adds `markFirstChunk`/`markFinished` wired via `onChunk`/`onFinish` for TTFT, TTLT, and throughput.
- Auto-counts `approvalPauses` on successful `create_workout`/`program_week` tool results.
- Leaves `totalCostUsd` undefined with an inline rationale (no reliable per-model rate across BYOK + house key).

**Enriched `aiToolCalls` (`convex/schema.ts`, `aiUsage.ts`, `ai/helpers.ts`)**
- Adds optional `runId`, `toolCallId`, `argsJson`, `resultPreview` plus a `by_runId` index.
- `withToolTracking` reads `runId` from `experimental_context`, `toolCallId` from `ToolExecutionOptions`, and bounded-stringifies args/results (4 KiB cap, circular-safe).

**PostHog AI removal (`convex/ai/coach.ts`, `convex/lib/posthog.ts`)**
- Drops `captureAiGeneration`; keeps `internal.aiUsage.record` and product analytics intact.

**Eval foundation**
- `convex/ai/evalScenarios.ts`: 13 scenarios tagged by capability, shared by `coachEvals.test.ts` + the Phoenix scripts.
- `scripts/evals/phoenix/`: `buildDataset.ts`, `runPromptSmoke.ts`, `runAgentExperiment.ts`, and `lib/{thresholds,report}.ts`. `runPromptSmoke.ts` soft-fails on fork PRs without a Google key.
- `convex/ai/evalHarness.ts`: prompt-only task today with a concrete TODO for the Convex-dispatch wiring needed for real end-to-end runs.

**CI + setup**
- `.github/workflows/ci.yml`: new `ai-smoke` PR job running `npm run ai:eval:smoke` with Phoenix + Google secrets.
- `.github/workflows/ai-evals-nightly.yml`: 04:17 UTC sync + experiment run.
- Package cleanup: added `@arizeai/phoenix-{otel,client,evals}` + `@opentelemetry/sdk-trace-node`; removed `@posthog/ai`, `@opentelemetry/core`, `exporter-trace-otlp-http`, `resources`, `sdk-trace-base`.
- Docs + setup prompts for `PHOENIX_API_KEY`/`PHOENIX_COLLECTOR_ENDPOINT`/`PHOENIX_PROJECT_NAME`/`PHOENIX_HOST`.

## Checklist

- [x] `npx tsc --noEmit` passes
- [x] `npm run lint` passes
- [x] `npm test` passes (1121 passed / 13 skipped; 6 new tests in `runTelemetry.test.ts`, full rewrite of `otel.test.ts`)
- [x] New logic has tests (happy path + at least one error/edge case)
- [x] No new or materially expanded file exceeds the 300-line soft cap
- [x] No file exceeds the 400-line hard cap
- [x] Functions stay near the 60-line convention
- [x] No new `any` types; no `as` casts outside existing boundaries
- [ ] Tested in browser (N/A — backend + scripts only)
- [x] Commits follow conventional format
- [x] No unused exports or dead code (`npm run knip` clean)

## Test Plan

**Automated**
- `npx tsc --noEmit`, `npm run lint`, `npm run knip`, `npm test` all green locally.
- New tests cover TTFT/TTLT math, throughput, idempotent first-chunk, approval-pause wiring, `totalCostUsd` invariant, and the Phoenix `startRunSpan`/`runInRunSpan`/`flushTelemetry` surface (zero-traceId fallback, callback error propagation).

**Manual (needs live Phoenix credentials; not done in this PR)**
1. Set `PHOENIX_API_KEY` on the Convex deployment and send one chat turn.
2. Confirm a single `roni.user_turn` trace exists in Phoenix Cloud with raw inputs/outputs, no secrets.
3. `db.query("aiRun").order("desc").take(1)` — `runId` matches the Phoenix `traceId`.
4. `db.query("aiToolCalls").filter(q => q.eq(q.field("runId"), <traceId>))` — row has `toolCallId`, `argsJson`, `resultPreview`.
5. Locally run `npm run ai:eval:smoke` with a Google key to confirm thresholds hold.

## Intentionally deferred

- Full Convex dispatch in the nightly harness (scaffolded with a concrete TODO). The prompt-only path runs in CI today.
- `totalCostUsd` computation — awaiting a per-model rate card.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * AI tracing moved to Phoenix Cloud; Phoenix-based observability and config added.
  * Shared AI evaluation framework: scenarios, rubrics, eval harness, prompt/agent smoke and nightly experiment runners.
  * Scripts to build/sync Phoenix datasets and run experiments.

* **Documentation**
  * Updated .env example, README, and contributing guide with Phoenix config and new AI eval commands.

* **Tests**
  * Added/updated tests for evals and telemetry helpers.

* **Chores**
  * New CI/workflows for PR smoke and nightly AI evals; added npm scripts and Phoenix-related dependencies.

### CODE REVIEW

File: convex/ai/otel.test.ts [hunk 1]
```diff
@@ -1,114 +1,105 @@
 import { describe, expect, it } from "vitest";
-import type { ReadableSpan } from "@opentelemetry/sdk-trace-base";
-import { resourceFromAttributes } from "@opentelemetry/resources";
-import { translateAttributes, translateSpan } from "./otel";
+import { flushTelemetry, isPhoenixEnabled, runInRunSpan, startRunSpan } from "./otel";
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/ai/otel.test.ts:2]: Fixed in f717e11. The assertion now mirrors the ambient `PHOENIX_API_KEY` (`Boolean(process.env.PHOENIX_API_KEY)`) so it passes identically in CI (secret set) and local dev (absent).

File: convex/ai/otel.ts [hunk 2]
```diff
@@ -3,96 +3,132 @@
...
+export async function runInRunSpan<T>(
+  meta: RunSpanMetadata,
+  fn: (handle: RunSpanHandle) => Promise<T>,
+): Promise<T> {
+  const { span, handle } = startRunSpan(meta);
+  try {
+    return await context.with(trace.setSpan(context.active(), span), () => fn(handle));
+  } finally {
+    await handle.end();
+  }
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/ai/otel.ts:145]: Fixed in f717e11. `runInRunSpan` now catches errors from the callback, calls `span.recordException(...)` and `span.setStatus({ code: SpanStatusCode.ERROR, message })` before rethrowing. The existing `finally` still ends and flushes the span.

File: convex/ai/resilience.ts [hunk 3]
```diff
@@ -228,19 +284,29 @@ async function attemptStream({
...
-    recordOutputs: false,
-    metadata: {
-      posthog_distinct_id: telemetry.userId,
-      posthog_trace_id: telemetry.runId,
-      threadId: telemetry.threadId,
-      provider: telemetry.provider,
-    },
+    recordInputs: true,
+    recordOutputs: true,
+    metadata,
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/ai/resilience.ts:281]: Holding on this one — raising to the author before reversing. The Phoenix migration spec locked raw capture as a hard requirement (Phase 4: "Capture all AI data and conversations in Phoenix: user messages, system instructions, training snapshots, message history sent to model, tool args/results, assistant outputs"). Secret sanitization is still enforced upstream: `byokErrors.ts` / `chatHelpers.ts` strip decrypted keys and Tonal tokens before anything reaches the SDK's telemetry layer. Happy to add an opt-in gate + pseudonymization helper if we decide to reverse the product decision.

File: scripts/evals/phoenix/buildDataset.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
+/**
+ * Sync shared eval scenarios into a Phoenix Cloud dataset so the Phoenix UI
+ * can run experiments against them.
+ *
+ *   PHOENIX_API_KEY=... npm run ai:dataset
+ *
+ * Reads PHOENIX_COLLECTOR_ENDPOINT (or PHOENIX_HOST) for the base URL and
+ * PHOENIX_PROJECT_NAME for labelling; falls back to sensible defaults.
+ */
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply scripts/evals/phoenix/buildDataset.ts:9]: Fixed in f717e11. `PHOENIX_PROJECT_NAME` is now read and threaded into the dataset description so operators can see which project the scenarios belong to.

File: scripts/evals/phoenix/lib/report.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  if (overall < OVERALL_PASS_RATE) {
+    reasons.push(
+      `overall pass rate ${overall.toFixed(2)} below threshold ${OVERALL_PASS_RATE.toFixed(2)}`,
+    );
+  }
+  for (const [cap, bucket] of rateByCapability(report)) {
+    const r = bucket.total === 0 ? 1 : bucket.passed / bucket.total;
+    const floor = CAPABILITY_PASS_RATE[cap];
+    if (r < floor) reasons.push(`${cap} pass rate ${r.toFixed(2)} below threshold ${floor.toFixed(2)}`);
+  }
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply scripts/evals/phoenix/lib/report.ts]: Fixed in f717e11. `decide` now iterates every entry in `CAPABILITY_PASS_RATE` and emits a failure reason when a capability has no scenarios or its pass rate falls below the floor. `approval_and_push` (currently zero scenarios) will now flag loudly until scenarios are added — catching exactly the silent-pass case you called out.

File: scripts/evals/phoenix/runAgentExperiment.ts [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+      }
+      const result = await runScenarioAgainstPrompt(scenario);
+      return {
+        text: result.text,
+        toolNames: result.toolNames,
+        modelId: result.modelId,
+      };
+    },
+    evaluators: [rubricEvaluator, bannedPhraseEvaluator, correctnessEvaluator],
+    concurrency: Number(process.env.PHOENIX_EXPERIMENT_CONCURRENCY ?? 2),
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply scripts/evals/phoenix/runAgentExperiment.ts]: Fixed in f717e11. Added a `positiveIntEnv()` helper that rejects NaN / ≤ 0 with a warning and falls back to the default; `PHOENIX_EXPERIMENT_CONCURRENCY` now goes through it.

File: scripts/evals/phoenix/runAgentExperiment.ts [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+  });
+
+  const failed = ran.failedRunCount;
+  const total = ran.exampleCount;
+  console.log(`Experiment ${ran.id}: ${total - failed}/${total} successful runs`);
+  if (failed > 0) process.exit(1);
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply scripts/evals/phoenix/runAgentExperiment.ts:182]: Deferring this. The nightly path is a scaffold today — it uses the prompt-only harness (see `evalHarness.ts` TODO for the Convex-dispatch plumbing that needs to land before real agent runs exist). Adding a `getExperiment(ran.id)` post-run scan to fail on evaluator `score === 0` is the right thing to wire, but it's coupled to having real experiment runs to grade. I'll add it in the follow-up that replaces `runScenarioAgainstPrompt` with the real Convex dispatch, rather than layering it on top of the scaffolded path.

File: scripts/evals/phoenix/runPromptSmoke.ts [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  EVAL_SCENARIOS,
+  type EvalScenario,
+  type Rubric,
+} from "../../../convex/ai/evalScenarios";
+import { BANNED_PHRASES, DEFAULT_MAX_RESPONSE_CHARS } from "./lib/thresholds";
+import { decide, printReport, type Report, type ScenarioResult } from "./lib/report";
+
+const MODEL_ID = process.env.PHOENIX_SMOKE_MODEL ?? "gemini-3-flash-preview";
+const CONCURRENCY = Number(process.env.PHOENIX_SMOKE_CONCURRENCY ?? 4);
+
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply scripts/evals/phoenix/runPromptSmoke.ts:36]: Fixed in f717e11. Same `positiveIntEnv()` helper as the experiment concurrency — rejects NaN / ≤ 0 so the `runAll` loop can't hang.

File: convex/ai/resilience.ts [hunk 9]
```diff
@@ -122,57 +91,88 @@ export async function streamWithRetry(
...
+            return { done: true };
+          }
+          if (!isTransientError(error)) {
+            const cls = errorClassName(error);
+            accumulator.setTerminalErrorClass(cls);
+            span.recordError(cls);
+            await reportError(ctx, { ...errorReport, error });
+            return { done: true };
+          }
+          return { done: false, error };
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/ai/resilience.ts:162]: Verified this isn't a real bug and leaving as-is. The convex-agent SDK already finalizes pending messages on every failed attempt via its own handlers:

- `onError: async (error) => { await call.fail(...) }` — `node_modules/@convex-dev/agent/dist/client/streamText.js:53-55`
- `onAsyncAbort: call.fail` — `node_modules/@convex-dev/agent/dist/client/streamText.js:34`

So transient-error branches rely on the SDK's cleanup; we don't strand pending rows. Also, this code path is pre-existing — the same return-without-cleanup shape existed before the Phoenix migration. And the suggested methods (`accumulator.finalizePending()` / `finalizeFailedAttempt()`) don't exist; calling `setTerminalErrorClass` on a transient that's about to retry would actively be wrong since it's not terminal.

File: scripts/evals/phoenix/runAgentExperiment.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (JeffOtano) [reply scripts/evals/phoenix/runAgentExperiment.ts:182]: yes
